pkg/providers/amifamily/ami.go

```diff
@@ -233,14 +268,16 @@ func getFiltersAndOwners(amiSelector map[string]string) ([]*ec2.Filter, []*strin
 	return filters, owners
 }
 
-func sortAMIsByCreationDate(amiRequirements map[AMI]scheduling.Requirements) []AMI {
-	amis := lo.Keys(amiRequirements)
-
+func sortAMIsByCreationDate(amis []AMI) []AMI {
 	sort.Slice(amis, func(i, j int) bool {
-		itime, _ := time.Parse(time.RFC3339, amis[i].CreationDate)
-		jtime, _ := time.Parse(time.RFC3339, amis[j].CreationDate)
-		return itime.Unix() >= jtime.Unix()
+		if amis[i].CreationDate != "" {
+			itime, _ := time.Parse(time.RFC3339, amis[i].CreationDate)
+			jtime, _ := time.Parse(time.RFC3339, amis[j].CreationDate)
+			return itime.Unix() >= jtime.Unix()
```

Comment:
What happens if they have the same Unix timestamp? Then we should have a predictable ordering by name